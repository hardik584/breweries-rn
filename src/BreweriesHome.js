import React, {useEffect, useState} from 'react';
import {
  ActivityIndicator,
  FlatList,
  Text,
  View,
  SafeAreaView,
  TouchableOpacity,
  TextInput,
  Image,
} from 'react-native';
import DelayInput from 'react-native-debounce-input';
import CustomStyle from './CustomStyle';
import BreweriesFlatList from './componants/BreweriesFlatList';


export default BreweriesHome = ({navigation}) => {
  const [isLoading, setLoading] = useState(true);
  const [data, setData] = useState([]);
  const [searchKeyword, setSearchKeyword] = useState('');

  const getBreweries = async () => {
    try {
      setLoading(true);
      var formattedUrl = 'https://api.openbrewerydb.org/breweries';
      if (searchKeyword !== '') {
        formattedUrl += '?by_city=' + searchKeyword;
      }
      console.log(formattedUrl);
      const response = await fetch(formattedUrl);
      const json = await response.json();
      setData(json);
    } catch (error) {
      console.error(error);
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    getBreweries();
  }, []);

  return (
    <SafeAreaView style={{flex: 1}}>
      <View style={CustomStyle.viewColumnStyle}>
        <View style={CustomStyle.viewRowStyle}>
          <TextInput
            style={CustomStyle.textInputStyle}
            placeholder={'Search Here'}
            onChangeText={text => {
              setSearchKeyword(text);
            }}
          />

          <TouchableOpacity
            style={CustomStyle.buttonStyle}
            activeOpacity={1}
            onPress={() => {
              getBreweries();
            }}>
            <Image
              source={{uri: 'https://img.icons8.com/ios-glyphs/344/search.png'}}
              style={CustomStyle.imageStyle}
            />
          </TouchableOpacity>
        </View>
        {isLoading ? (
          <ActivityIndicator />
        ) : data.length == 0 ? (
          <Text style={[CustomStyle.baseText, {textAlign: 'center'}]}>
            {' '}
            No record found!{' '}
          </Text>
        ) : (
          <BreweriesFlatList data={data} navigation={navigation} />
        )}
      </View>
    </SafeAreaView>
  );
};
