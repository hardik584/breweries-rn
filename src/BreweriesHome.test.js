import BreweriesHome from './BreweriesHome';
import renderer from 'react-test-renderer';
import React, {useState, useEffect} from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render, screen, fireEvent} from '@testing-library/react-native';

import AppNavigator from '../App';



describe('Checking Breweries Home Rendering', () => {
  const fakeNavigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
  };

  it(`Checking Breweries Home Rendering`, () => {
    fetch = jest.fn(() => Promise.resolve());
    const tree = renderer.create(<BreweriesHome> </BreweriesHome>);
    expect(tree).toMatchSnapshot();
  });

  it(`Checking Breweries Home with fakeNavigation Rendering`, () => {
    const tree = renderer
      .create(<BreweriesHome navigation={fakeNavigation}> </BreweriesHome>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

//   test('Enter search text to search', () => {
//     global.fetch = jest.fn();
//     fetch.mockResponseIsSuccess = (body) => {
//       fetch.mockImplementationForOnce (
//         () => Promise.resolve({json: () => Promise.resolve(JSON.parse(body))})
//       );
//     };
//     const { getByPlaceholderText, getByText, getAllByText, findBy } = render(
//       <BreweriesHome />
//     );
  
//     fireEvent.changeText(getByPlaceholderText('Search Here'),'USA');
//   // fireEvent.press(getByText('Add Item'));

//   // fireEvent.changeText(getByPlaceholderText('Enter grocery item'),'Mango');
//   // fireEvent.press(getByText('Add Item'));

//   const searchList = getAllByText('USA');
  
//   expect(groceryList).toHaveLength(1); 
// });

