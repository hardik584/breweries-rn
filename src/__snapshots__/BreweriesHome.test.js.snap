// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Breweries Home Rendering Checking Breweries Home Rendering 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "padding": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 5,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Search Here"
        style={
          Object {
            "borderColor": "#000000",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "height": 50,
            "padding": 10,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#38b6ba",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginLeft": 5,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://img.icons8.com/ios-glyphs/344/search.png",
            }
          }
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
    <ActivityIndicator />
  </View>
</RCTSafeAreaView>
`;

exports[`Checking Breweries Home Rendering Checking Breweries Home with fakeNavigation Rendering 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "padding": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 5,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Search Here"
        style={
          Object {
            "borderColor": "#000000",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "height": 50,
            "padding": 10,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#38b6ba",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginLeft": 5,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://img.icons8.com/ios-glyphs/344/search.png",
            }
          }
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
    <ActivityIndicator />
  </View>
</RCTSafeAreaView>
`;
